Fix BodyParamsMiddleware: Valid JSON containing a scalar value causes InvalidArgumentException

psfpro opened the pull request:
|    Q          |   A
|-------------- | ------
| Bugfix        | yes
| BC Break      | yes

### Description
 Fix for https://github.com/mezzio/mezzio-helpers/issues/1
 Based on the documentation, we see "a null value if nothing was available to parse".
 
```
     * The data IS NOT REQUIRED to come from $_POST, but MUST be the results of
     * deserializing the request body content. Deserialization/parsing returns
     * structured data, and, as such, this method ONLY accepts arrays or objects,
     * or a null value if nothing was available to parse.
 ```
 https://www.php-fig.org/psr/psr-7/
 
 In this case we got unavailable data for converting to array

psfpro committed:
Fix BodyParamsMiddleware: Valid JSON containing a scalar value causes InvalidArgumentException

lcobucci commented:
This package now requires PHP 7.3+, a better solution is to rely on JSON_THROW_ON_ERROR. https://www.php.net/manual/en/json.constants.php

Edit: sorry, I'm talking about a different bit of the code - and a different improvement as well. Just ignore me, I'm a teapot.

lcobucci commented:
Can you please provide some tests, though? 😬

weierophinney committed:
qa: add unit tests for fix to parsed body

- Adds unit tests to ensure that a JSON value that does not resolve to an array or map passes a `null` value to the request `withParsedBody()` method.
- Applies CS fixes.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: rename phpcs config file with .dist extension

Mainly to trigger GHA to run.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney commented:
I've added tests for this behavior and pushed them, which has picked up the new GHA workflow to validate them.

While this is technically a BC break, we were also in violation of the PSR-7 specification with our previous behavior, so I consider this a bugfix, as our previous behavior would lead to results unexpected within a PSR-7 context.
